version: '3.8'

services:
  db:
    image: postgres:15-alpine
    container_name: botdb_container
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: botdbcont
      LC_ALL: en_US.UTF-8
      LANG: en_US.UTF-8
      POSTGRES_HOST_AUTH_METHOD: scram-sha-256
    ports:
      - "6432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres -d botdbcont || exit 1"]
      interval: 5s
      timeout: 5s
      retries: 30
    networks:
      - bot_network

  bot:
    build: .
    depends_on:
      db:
        condition: service_healthy
    ports:
      - "5000:80"
    environment:
      - ConnectionStrings=Host=db;Database=botdbcont;Username=postgres;Password=postgres;Port=5432
    networks:
      - bot_network
    restart: on-failure

volumes:
  postgres_data:

networks:
  bot_network:
    driver: bridge